core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -398,8 +398,11 @@ class BrokerMetadataListener(
         build()
       try {
         _image.write(writer, options)
-      } finally {
-        writer.close()
+        writer.close(true)
```

Comment:
`MetadataImage::write` calls `ImageWriter::close(true)`. Can we remove that call? It if very difficult to read and understand code that closes `AutoCloseable` in different layers of the code.

Reply:
Yes, you are right that we're already calling `ImageWriter.close(true)` in `MetadataImage.write`